### PR TITLE
feat: add rail viewed events to home view section

### PR DESCRIPTION
**Pull request by MounirDhahri**
This PR resolves [ONYX-1297] 

> [!NOTE]
> I suggest you hide white spaces for easier reviewing

### Description

This PR adds `railViewed` tracking events to the new home view. 
I initially tried to implement the same behavior using built-in `FlatList` viewability config pairs. However that meant that we would need to trigger an additional query to tell the count of items in a section.

We eventually settled for a solution that is similar to what we do in Web. 
Now we have a new sentinel component, just as in Force, that can tell when a component is in view port. I added that as a shared component and created an additional one using it for the new home view sections.

https://github.com/user-attachments/assets/528c429d-a2c6-4472-986e-5befd6b6f580

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- add rail viewed events to home view section - mounir

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[ONYX-1297]: https://artsyproduct.atlassian.net/browse/ONYX-1297?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by dblandin — COMMENTED** [review]
Awesome! This looks great. Added one comment re: adding the dependency vs adopting the underlying code. Curious what your thoughts are on that.

File: src/app/Scenes/HomeView/HomeViewContext.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+  addTrackedSection: action((state, payload) => {
+    if (state.trackedSections.includes(payload)) {
+      return
+    }
+    state.trackedSections.push(payload)
+  }),
+}
+
+export const HomeViewStore = createContextStore(HomeViewStoreModel)
```
[the remark this answers is not in the document]
Comment (dblandin) [reply src/app/Scenes/HomeView/HomeViewContext.ts:20]: With easy-peasy, when is this data reset/refreshed? When the app launches or comes into the foreground?

File: package.json [hunk 2]
```diff
@@ -135,6 +135,7 @@
     "@shopify/flash-list": "1.6.4",
     "@stripe/stripe-react-native": "0.36.0",
     "@styled-system/theme-get": "5.1.2",
+    "@svanboxel/visibility-sensor-react-native": "1.0.2",
```
[the remark this answers is not in the document]
Comment (MounirDhahri) [reply package.json]: good idea- that makes sense. I will do it and add a reference to the original author

[the remark this answers is not in the document]
Comment (MounirDhahri) [reply package.json]: Here it is
https://github.com/artsy/eigen/pull/10816/commits/4462d1e1158b6d2f2c471770fa6b01b6e450b0b8

**Review by dblandin — APPROVED** [review]

**Pull request merged**